### PR TITLE
[Silabs] [Wi-Fi] [SiWx917] Added support for new SDK 3.1.1 for 917 NCP 

### PR DESCRIPTION
**Pull request by shgutte**
Description of Problem/Feature:
Added support for new SDK 917 NCP as well  as  gsdk 4.4.0 

Description of Fix/Solution:
Added support for new SDK 917 NCP as well  as  gsdk 4.4.0 for for board 4186C and 4197C 

Testing Done:
917 NCP + 4186C
917 NCP + 4187C

### CODE REVIEW

File: third_party/silabs/efr32_sdk.gni [hunk 1]
```diff
@@ -130,6 +130,7 @@ template("efr32_sdk") {
     # Treat these includes as system includes, so warnings in them are not fatal.
     _include_dirs = [
       "${efr32_sdk_root}",
+      "${examples_plat_dir}",
```
Comment (jmartinez-silabs) [inline third_party/silabs/efr32_sdk.gni:133]: This is strange and is not part of the efr32_sdk. what are we including this here?

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp [hunk 2]
```diff
@@ -78,15 +84,26 @@ using namespace ::chip::DeviceLayer::Internal;
 
 void sl_ble_init()
 {
-    uint8_t randomAddrBLE[6] = { 0 };
+    uint8_t randomAddrBLE[RSI_BLE_ADDR_LENGTH] = { 0 };
+#if SIWX_917
+    sl_status_t sl_status;
+    //! Get Random number of desired length
+    sl_status = sl_si91x_trng_get_random_num((uint32_t *)randomAddrBLE, RSI_BLE_ADDR_LENGTH);
```
Comment (jmartinez-silabs) [inline src/platform/silabs/rs911x/BLEManagerImpl.cpp:91]: this work is already done in another PR isn't it?  It is a bit confusing to have the same additions in 2 pr

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.c [hunk 3]
```diff
@@ -31,10 +31,10 @@
 #include "FreeRTOS.h"
 #include "event_groups.h"
 #include "task.h"
-
 #include "wfx_host_events.h"
 #include "wfx_rsi.h"
 
+
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/efr32/rs911x/wfx_rsi_host.c:37]: No changes in this file really should probably be left as it was

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c [hunk 4]
```diff
@@ -301,6 +364,25 @@ static sl_status_t wfx_rsi_init(void)
         SILABS_LOG("sl_wifi_get_mac_address failed: %x", status);
         return status;
     }
+#ifdef SIWX_917
+    const uint32_t trngKey[TRNGKEY_SIZE] = { 0x16157E2B, 0xA6D2AE28, 0x8815F7AB, 0x3C4FCF09 };
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c:368]: same comment: this code is added in another PR that I reviewed and is not needed to enable this PR. You should not be replicating this work again.

File: examples/platform/silabs/efr32/rs911x/hal/efx32_ncp_host.c [hunk 5]
```diff
@@ -0,0 +1,287 @@
+/*******************************************************************************
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/efr32/rs911x/hal/efx32_ncp_host.c:1]: isn't that file in wifi_sdk ?

File: examples/platform/silabs/efr32/rs911x/hal/sl_si91x_ncp_utility.c [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+#if SL_LCDCTRL_MUX
+
+/********************************************************
+ * @fn   sl_wfx_host_pre_lcd_spi_transfer(void)
+ * @brief
+ *        Deal with the SPI PINS MUX that are associated with LCD. for setup before data transfer PINS.
+ * @return
+ *        None
+ **********************************************************/
+sl_status_t sl_wfx_host_pre_lcd_spi_transfer(void)
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/efr32/rs911x/hal/sl_si91x_ncp_utility.c:80]: isn't the majority of this code already in efx_spi.c?

**Review by jmartinez-silabs — CHANGES_REQUESTED** [review]

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c [hunk 7]
```diff
@@ -41,6 +41,16 @@
 
 #include "ble_config.h"
 
+#if SL_ICD_ENABLED && SIWX_917
```
Comment (chirag-silabs) [inline examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c:44]: These changes were already added as part of my PR. Rebase the PR these should be removed

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c [hunk 8]
```diff
@@ -198,6 +213,44 @@ sl_status_t join_callback_handler(sl_wifi_event_t event, char * result, uint32_t
 }
 
 #if SL_ICD_ENABLED
+
+#if SIWX_917
```
Comment (chirag-silabs) [inline examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c:217]: Same here

File: third_party/silabs/efr32_sdk.gni [hunk 9]
```diff
@@ -204,11 +205,21 @@ template("efr32_sdk") {
...
+      "${efr32_sdk_root}/platform/Device/SiliconLabs/EFR32MG24/Include",
+      ]
+    }
+
     if (use_SiWx917) {
       _include_dirs += [
-        "${wifi_sdk_root}/components/si91x/inc",
         "${wifi_sdk_root}/components/si91x/memory",
-        "${wifi_sdk_root}/components/si91x/sl_net/inc",
+        "${wifi_sdk_root}/components/device/silabs/si91x/wireless/inc",
```
Comment (chirag-silabs) [inline third_party/silabs/efr32_sdk.gni:217]: these should be added in rs9117.gni file right
There is no point of adding this in multiple files

File: src/platform/silabs/rs911x/rsi_ble_config.h [hunk 10]
```diff
@@ -35,12 +36,21 @@
 #define RSI_FAILURE -1
 #endif
 
+#define WIFI_ENABLE_SECURITY_WPA3_TRANSITION 0
```
Comment (chirag-silabs) [inline src/platform/silabs/rs911x/rsi_ble_config.h:39]: Why we are defining this to 0 directly

File: src/platform/silabs/rs911x/rsi_ble_config.h [hunk 11]
```diff
@@ -35,12 +36,21 @@
 #define RSI_FAILURE -1
 #endif
 
+#define WIFI_ENABLE_SECURITY_WPA3_TRANSITION 0
+
+#ifndef CHIP_DEVICE_CONFIG_ENABLE_IPV4
```
Comment (chirag-silabs) [inline src/platform/silabs/rs911x/rsi_ble_config.h:41]: here as well

File: third_party/silabs/efr32_sdk.gni [hunk 12]
```diff
@@ -631,7 +652,7 @@ template("efr32_sdk") {
       "${efr32_sdk_root}/platform/emlib/src/em_timer.c",
       "${efr32_sdk_root}/platform/peripheral/src/peripheral_sysrtc.c",
       "${efr32_sdk_root}/platform/radio/rail_lib/plugin/pa-conversions/pa_conversions_efr32.c",
-      "${efr32_sdk_root}/platform/radio/rail_lib/plugin/rail_util_power_manager_init/sl_rail_util_power_manager_init.c",
+   #   "${efr32_sdk_root}/platform/radio/rail_lib/plugin/rail_util_power_manager_init/sl_rail_util_power_manager_init.c",
```
[the remark this answers is not in the document]
Comment (shgutte) [reply third_party/silabs/efr32_sdk.gni]: Uncommented the file

File: third_party/silabs/efr32_sdk.gni [hunk 13]
```diff
@@ -204,11 +205,21 @@ template("efr32_sdk") {
       "${sl_ot_efr32_root}",
     ]
 
+    if (silabs_family == "efr32mg24") {
```
[the remark this answers is not in the document]
Comment (shgutte) [reply third_party/silabs/efr32_sdk.gni:208]: Yes but it's not taking the included files for board 4186C we tried a lot of times, and if we separate it it works fine otherwise we are getting build error for this files if you want i can show you the issue

File: third_party/silabs/efr32_sdk.gni (hunk 1 above)
Comment (shgutte) [reply third_party/silabs/efr32_sdk.gni:133]: we are trying to include spi_multiplex.h from that location

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 2 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp]: removed

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c (hunk 4 above)
Comment (shgutte) [reply examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c]: rebased the branch

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 2 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp]: rebased the branch

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.c (hunk 3 above)
Comment (shgutte) [reply examples/platform/silabs/efr32/rs911x/wfx_rsi_host.c:37]: it's same file i don't know why it's showing + line

File: examples/platform/silabs/efr32/rs911x/hal/efx32_ncp_host.c (hunk 5 above)
Comment (shgutte) [reply examples/platform/silabs/efr32/rs911x/hal/efx32_ncp_host.c:1]: Yes we have updated it for SPI multiplexing changes so copied till they add the API for SPI multiplexing

File: examples/platform/silabs/efr32/rs911x/hal/sl_si91x_ncp_utility.c (hunk 6 above)
Comment (shgutte) [reply examples/platform/silabs/efr32/rs911x/hal/sl_si91x_ncp_utility.c:80]: Yeah we will create this so both efx_spi as well as efx32_ncp_host files can you have this utility function going further soon we will remove it from efx_spi file and use this file for definitions

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c (hunk 7 above)
Comment (shgutte) [reply examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c]: rebased

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c (hunk 8 above)
Comment (shgutte) [reply examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c]: rebased

File: examples/platform/silabs/efr32/rs911x/sl_wlan_config.h [hunk 14]
```diff
@@ -34,7 +34,7 @@ static const sl_wifi_device_configuration_t config = {
     .boot_config = { .oper_mode = SL_SI91X_CLIENT_MODE,
```
Comment (chirag-silabs) [inline examples/platform/silabs/efr32/rs911x/sl_wlan_config.h:34]: this file can be used directly right examples/platform/silabs/SiWx917/SiWx917/sl_wlan_config.h?

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c (hunk 4 above)
Comment (shgutte) [reply examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c]: rebased the branch

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c (hunk 7 above)
Comment (shgutte) [reply examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c]: rebased the brach

File: src/platform/silabs/rs911x/rsi_ble_config.h (hunk 11 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/rsi_ble_config.h]: removed

File: third_party/silabs/efr32_sdk.gni (hunk 9 above)
Comment (shgutte) [reply third_party/silabs/efr32_sdk.gni]: Added files in efr32_sdk.gni only if you add it int rs9117.gni it's giving build error for some include files

File: examples/platform/silabs/efr32/rs911x/sl_wlan_config.h (hunk 14 above)
Comment (shgutte) [reply examples/platform/silabs/efr32/rs911x/sl_wlan_config.h:34]: Going further may configuration so we are keeping it separate right now

File: third_party/silabs/efr32_sdk.gni (hunk 1 above)
Comment (jmartinez-silabs) [reply third_party/silabs/efr32_sdk.gni:133]: Maybe spi_multiplex.h and related code should not be in `examples_plat_dir` if it is needed by SDK files.

the dependency should be the other way around (examples_plat_dir sources requiring the SDK. not SDK requiring examples_plat_dir)

Please add a TODO comment explain why it is included three and that it need to be fixed/cleaned

File: third_party/silabs/efr32_sdk.gni (hunk 13 above)
[the remark this answers is not in the document]
Comment (jmartinez-silabs) [reply third_party/silabs/efr32_sdk.gni:208]: Sounds odd. There is something else at play here.

File: examples/platform/silabs/efr32/BUILD.gn [hunk 15]
```diff
@@ -281,7 +281,9 @@ source_set("efr32-common") {
   }
 
   if (chip_enable_wifi) {
-    if (use_rs9116 || use_SiWx917) {
+
+
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/efr32/BUILD.gn:285]:
```suggestion
```

File: examples/platform/silabs/efr32/rs911x/wfx_rsi_host.c (hunk 3 above)
Comment (jmartinez-silabs) [inline examples/platform/silabs/efr32/rs911x/wfx_rsi_host.c:37]:
```suggestion
```

File: src/platform/silabs/efr32/wifi/wfx_notify.cpp [hunk 16]
```diff
@@ -43,6 +43,7 @@
 #include <app/server/Dnssd.h>
 #include <app/server/Server.h>
 
+
```
Comment (jmartinez-silabs) [inline src/platform/silabs/efr32/wifi/wfx_notify.cpp:46]:
```suggestion
```

File: third_party/silabs/efr32_sdk.gni [hunk 17]
```diff
@@ -223,9 +236,13 @@ template("efr32_sdk") {
 
         # network_manager component
         "${wifi_sdk_root}/components/service/network_manager/inc",
+
```
Comment (jmartinez-silabs) [inline third_party/silabs/efr32_sdk.gni:239]:
```suggestion
```

File: third_party/silabs/efr32_sdk.gni [hunk 18]
```diff
@@ -602,6 +626,7 @@ template("efr32_sdk") {
 
   source_set(sdk_target_name) {
     sources = [
+
```
Comment (jmartinez-silabs) [inline third_party/silabs/efr32_sdk.gni:629]:
```suggestion
```

File: third_party/silabs/efr32_sdk.gni (hunk 13 above)
[the remark this answers is not in the document]
Comment (shgutte) [reply third_party/silabs/efr32_sdk.gni:208]: I'll raise JIRA on this to investigate further

File: third_party/silabs/efr32_sdk.gni (hunk 1 above)
Comment (shgutte) [reply third_party/silabs/efr32_sdk.gni:134]: Added TODO

**Review by jmartinez-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]
